### PR TITLE
list:strict_zip

### PR DESCRIPTION
**Pull request by bsnyder788**

### CODE REVIEW

File: gleam_stdlib/src/list.gleam [hunk 1]
```diff
@@ -4,6 +4,9 @@ pub enum Empty =
 pub enum NotFound =
   | NotFound
 
+pub enum ListLengthMismatch =
+  | ListLengthMismatch
```
Comment (lpil) [inline gleam_stdlib/src/list.gleam]: Tiny nitpick- can we call this `LengthMismatch`? Thank you.

**Comment by lpil** [comment]
Would you mind updating the changelog too? May need to rebase to fix the conflicts.

File: gleam_stdlib/src/list.gleam (hunk 1 above)
Comment (bsnyder788) [reply gleam_stdlib/src/list.gleam]: sure!

**Comment by bsnyder788** [comment]
@lpil should be rebased now, and the other issues have been take care of.

File: gleam_stdlib/test/list_test.gleam [hunk 2]
```diff
@@ -204,6 +204,23 @@ pub fn zip_test() {
   |> expect:equal(_, [{5, 1}, {6, 2}])
 }
 
+pub fn srict_zip_test() {
```
Comment (lpil) [inline gleam_stdlib/test/list_test.gleam]: Oops, typo here :)

Comment (bsnyder788) [reply gleam_stdlib/test/list_test.gleam]: fixed 👍

**Comment by lpil** [comment]
Thank you!!

**Pull request merged**